mlrun/model_monitoring/application.py

```diff
@@ -284,17 +260,35 @@ def __init__(
         self.output_stream = None
         self.name = name or "PushToMonitoringWriter"
 
-    def do(self, event: list[ModelMonitoringApplicationResult]) -> None:
+    def do(self, event: Tuple[list[ModelMonitoringApplicationResult], dict]) -> None:
         """
         Push application results to the monitoring writer stream.
 
-        :param event: Monitoring result(s) to push.
+        :param event: Monitoring result(s) to push and the original event from the controller.
         """
         self._lazy_init()
-        for result in event:
-            data = result.to_dict()
-            logger.info(f"Pushing data = {data} \n to stream = {self.stream_uri}")
-            self.output_stream.push([data])
+        application_results, application_event = event
+        metadata = {
+            mm_constant.WriterEvent.APPLICATION_NAME: application_event[
+                mm_constant.ApplicationEvent.APPLICATION_NAME
+            ],
+            mm_constant.WriterEvent.ENDPOINT_ID: application_event[
+                mm_constant.ApplicationEvent.ENDPOINT_ID
+            ],
+            mm_constant.WriterEvent.START_INFER_TIME: application_event[
+                mm_constant.ApplicationEvent.START_INFER_TIME
+            ],
+            mm_constant.WriterEvent.END_INFER_TIME: application_event[
+                mm_constant.ApplicationEvent.END_INFER_TIME
+            ],
+            mm_constant.WriterEvent.CURRENT_STATS: json.dumps(
+                application_event[mm_constant.ApplicationEvent.CURRENT_STATS]
+            ),
+        }
+        for result in application_results:
+            result.to_dict().update(metadata)
+            logger.info(f"Pushing data = {result} \n to stream = {self.stream_uri}")
+            self.output_stream.push([result])
```

Comment:
I still don't see how it would work. You update a dictionary object returned by `result.to_dict()`, but pass `result` which is not the same object.

I kindly ask you to run the system test and request a review only after it passes.